source/Coop.Core/Client/States/MainMenuState.cs

```diff
@@ -35,6 +36,7 @@ public override void Connect()
     internal void Handle_NetworkConnected(MessagePayload<NetworkConnected> obj)
     {
         Logic.ValidateModules();
+        messageBroker.Publish(this, new SendInformationMessage("A new player is joining the game, pausing"));
```

Comment:
Instead create a JoinMessage service that subscribes to the same events. This is separate functionality than the states here